### ISSUE TITLE
Docker, Core system name SERVICEREGISTRY is not recognized.

### ISSUE THREAD

**Issue by uessrodriguez** [eclipse-arrowhead/core-java-spring#361]
Hello

Trying deploy a docker enviroment from docker path directory https://github.com/eclipse-arrowhead/core-java-spring/tree/master/docker

Following the steps.

Plus: I've removed a space at the end, in file 
https://github.com/eclipse-arrowhead/core-java-spring/blob/master/docker/core_system_config/serviceregistry.properties
in line core_system_name=SERVICEREGISTRY 

With all AMD versions, with database created and filled

I can't start the SERVICEREGISTRY

```
main] e.a.c.s.ServiceRegistryMain              : Starting ServiceRegistryMain v4.3.0 on 53927d823100 with PID 9 (/serviceregistry/arrowhead-serviceregistry.jar started by root in /serviceregistry)
2021-11-15 11:01:30.050  INFO 53927d823100 --- [           main] e.a.c.s.ServiceRegistryMain              : No active profile set, falling back to default profiles: default
2021-11-15 11:01:58.791  WARN 53927d823100 --- [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'serviceRegistryApplicationInitListener': Unsatisfied dependency expressed through field 'coreSystemRegistrationProperties'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'coreSystemRegistrationProperties': Invocation of init method failed; nested exception is java.util.ServiceConfigurationError: Core system name SERVICEREGISTRY is not recognized.
2021-11-15 11:01:58.982 ERROR 53927d823100 --- [           main] o.s.b.SpringApplication                  : Application run failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'serviceRegistryApplicationInitListener': Unsatisfied dependency expressed through field 'coreSystemRegistrationProperties'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'coreSystemRegistrationProperties': Invocation of init method failed; nested exception is java.util.ServiceConfigurationError: Core system name SERVICEREGISTRY is not recognized.
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:596)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:90)
```

Others container runs well, ORCHESTRATOR as example

```
2021-11-15 11:02:09.386  INFO 38669133a74d --- [           main] e.a.c.ApplicationInitListener            : Core system name: ORCHESTRATOR
2021-11-15 11:02:09.387  INFO 38669133a74d --- [           main] e.a.c.ApplicationInitListener            : Server mode: NOT SECURED
```

But all infraestructure depends of SERVICEREGISTRY :(

What I can be doing wrong? 

TIA

**Comment by tsvetlin** [comment]
Hello!

Could you solve the issue, or the issue is still relevant?

**Comment by uessrodriguez** [comment]
Yes, it's solved compiling and using the jar created in docker.

Thanks

**Comment by tsvetlin** [comment]
If I understand correctly, did you build your own docker image?

**Comment by uessrodriguez** [comment]
Really we change the Jar in the docker image, but yes, we build a new docker image changing a bit the docker compose.

Thanks.

**Comment by tsvetlin** [comment]
If your docker compose is public, could you please share it, so I can update the images?

**Comment by uessrodriguez** [comment]
HI, its not public :( sorry, but the only change was: add another entry in volumen section with link to our compiled JAR.

As you see is a fix to use our compiled JAR, I Think that Service JAR SERVICEREGISTRY in docker image is wrong or something...

**Comment by dorinelfilip** [comment]
I have run into the same problem.
Recompiling the jar and recreating the Docker image with the new asset was the solution for me.

**Comment by oyuz** [comment]
Yeah the docker setup does not work out of the box.. getting the exact same issue on a fresh windows setup. If anyone could give a detailed explanation to someone who does not have a lot of experience with docker I would highly appreciate it

**Comment by esen96** [comment]
@dorinelfilip Would you mind giving a more detailed description on how you did it? I have the same issue and am new to docker so I don't understand the steps you took to solve it

**Comment by hebomstr** [comment]
@esen96 I managed to resolve this issue by changing the core_system_config/serviceregistry.properties file from

```
core_system_name=SERVICEREGISTRY
```

into

```
core_system_name=SERVICE_REGISTRY
```

Seems to be a typo in the properties file. Hope this helps.

**Comment by borditamas** [comment]
Hello,

we have just released the v4.6.1, where a more flexible docker approach has been taken and bug fixes were managed.
[Take a look](https://github.com/eclipse-arrowhead/core-java-spring/wiki/Docker) and please open a new issue if you still runs into some error.
